pkg/controller/cluster/main-controller.go

```diff
@@ -423,25 +461,27 @@ func (c *Controller) Start(threadiness int, stopCh <-chan struct{}) error {
 				}
 			} else {
 				klog.Infof("Console TLS is not enabled")
-				close(consoleTLS)
+				wg.Done()
 			}
 		}()
 
-		klog.Info("Waiting for API to start")
-		<-apiServerWillStart
-
-		klog.Info("Waiting for Upgrade Server to start")
-		<-upgradeServerWillStart
+		if IsSTSEnabled() {
+			wg.Add(1)
+			go func() {
+				klog.Infof("STS is enabled, starting STS API certificate setup")
+				c.generateSTSTLSCert()
+				wg.Done()
```

Comment:
better to `defer wg.Done()` at the beginning of the goroutine

Reply:
done